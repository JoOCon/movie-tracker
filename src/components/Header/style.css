.header {
  background-color: #116466;
  display: flex;
  justify-content: space-between;
  padding: 0 1em; }

.title-container {
  display: flex;
  flex-grow: 1;
  justify-content: center; }

.header h1 {
  color: #ffcb9a;
  margin: .5em 0;
  font-size: 3em; }
