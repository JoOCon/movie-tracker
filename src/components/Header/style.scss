@import '../../styles/variables.scss';
@import '../../styles/mixins.scss';


.header {
  background-color: $turquoise;
  border-bottom: 1px solid $gold;
  display: flex;
  justify-content: space-between;
  padding: 1em 1em;
}

.title-container {
  display: flex;
  flex-grow: 1;
  justify-content: left;
}

.title {
  color: $gold;
  font-family: $title-font;
  font-size: 3.5em;
  margin: .5em 0;
  text-shadow: 2.5px 2.5px black;
}

.home-redirect {
  @include text-link;
  margin-right: 1em;
}

.account-redirect {
  @include text-link;
  margin-left: 1em;
}

.button-wrap {
  display: flex;
  justify-content: space-between;
  width: 100%;
}