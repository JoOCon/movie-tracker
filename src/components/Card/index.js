import React from 'react';
import './style.css';
import StarRatings from 'react-star-ratings';

export const Card = ({ poster_path, vote_average }) => {
  const rating = vote_average / 2;
  return (
    <div className="card">
      <img src={`https://image.tmdb.org/t/p/w200${poster_path}`} alt='movie data' />
      <StarRatings starDimension={'1em'} rating={rating} numberOfStars={5} starRatedColor={'gold'} />
    </div>
  );
};
