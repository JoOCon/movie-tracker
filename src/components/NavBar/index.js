import React from 'react';
import { Link } from 'react-router-dom';
import './style.css';

export const NavBar = () => {
  return (
    <div className='nav-bar'>
      <Link to='/login'><button className='login'>Login</button></Link>
      <Link to='/signup'><button className='signup'>SignUp</button></Link>
    </div>
  );
};
