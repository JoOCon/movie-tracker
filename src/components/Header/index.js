import React from 'react';
import { NavBar } from '../../containers/NavBar';
import { Route } from 'react-router-dom';

export const Header = () => {
  return (
    <div>
      <h1>Movie Tracker</h1>
      <Route path='/' component={NavBar} />
    </div>
  );
};
