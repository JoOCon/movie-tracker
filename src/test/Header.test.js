import React from 'react';
import { Header } from '../components/Header';
import { shallow } from 'enzyme';

describe('Header tests', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  it('renders the Card with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
