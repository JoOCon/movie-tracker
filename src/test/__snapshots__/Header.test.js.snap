// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests renders the Card with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title-container"
>
          <h1
                    className="title"
          >
                    Movie Tracker
          </h1>
</div>,
        <Route
          component={[Function]}
          path="/"
/>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="title"
>
            Movie Tracker
</h1>,
          "className": "title-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Movie Tracker",
            "className": "title",
          },
          "ref": null,
          "rendered": "Movie Tracker",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title-container"
>
            <h1
                        className="title"
            >
                        Movie Tracker
            </h1>
</div>,
          <Route
            component={[Function]}
            path="/"
/>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="title"
>
              Movie Tracker
</h1>,
            "className": "title-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Movie Tracker",
              "className": "title",
            },
            "ref": null,
            "rendered": "Movie Tracker",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
